packages/core/src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -230,7 +230,11 @@ export default class StripeUPEPaymentStrategy implements PaymentStrategy {
         try {
             return await this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
         } catch (error) {
-            return await this._processAdditionalAction(error);
+            if (error instanceof Error) {
+                return await this._processAdditionalAction(error);
+            }
+
+            return Promise.reject(this._store.getState());
```

Comment:
Perhaps more appropriate to rethrow the error instead?

```suggestion
            throw error;
```